client/my-sites/theme/main.jsx

```diff
@@ -1303,8 +1335,14 @@ class ThemeSheet extends Component {
 			plansUrl = localizeUrl( 'https://wordpress.com/pricing' );
 		} else if ( siteSlug ) {
 			const redirectTo = `/theme/${ themeId }${ section ? '/' + section : '' }/${ siteSlug }`;
-			const plan = isExternallyManagedTheme || isBundledSoftwareSet ? PLAN_BUSINESS : PLAN_PREMIUM;
+			let plan;
+			if ( ! isThemeAllowedOnSite ) {
+				plan = THEME_TIERS[ themeTier.slug ].minimumUpsellPlan;
+			} else {
```

Comment:
Once the new tiers are rolled out, this should be the only logic needed to determine the plan.